source/grid/manifold.cc

```diff
@@ -659,16 +654,20 @@ add_new_points (const std::vector<Point<spacedim> > &surrounding_points,
           if (new_point[d] < 0)
             new_point[d] += periodicity[d];
 
-      new_points.push_back(project_to_manifold(surrounding_points, new_point));
+      // TODO should this use surrounding_points_start or surrounding_points?
+      // The older version used surrounding_points
+      new_points[row] = project_to_manifold(make_array_view(surrounding_points.begin(),
+                                                            surrounding_points.end()),
+                                            new_point);
```

Comment:
I am not sure; it has been long enough since I wrote this that I don't even really remember the issue. I will look at this with @luca-heltai at some point during the next few days.